src/core/files.ml

```diff
@@ -210,8 +210,7 @@ let new_lib_mapping : string -> unit = fun s ->
   let file_path = Filename.realpath file_path in
   let new_mapping =
     try ModMap.add module_path file_path !lib_mappings
-    with ModMap.Already_mapped ->
-    fatal_no_pos "Module path [%a] is already mapped." Path.pp module_path
+    with ModMap.Already_mapped -> !lib_mappings
```

Comment:
It means that the exception Already_mapped becomes useless to raise in the code of add. It would be better to change the code of add instead. The implications are not clear though.

Reply:
Could you please take advantage of this to fix the typo in the comment on line 70 ("the the")?

Reply:
@ejgallego made some remark about this part of the code in https://github.com/Deducteam/lambdapi/pull/434#issuecomment-712387097 . @gabrielhdt do you have a better understanding ?

Reply:
In https://github.com/Deducteam/lambdapi/pull/434#issuecomment-713365488, @gabrielhdt said:

> I think I found out why, and I fixed it in `dune2`, see the description of #449: same module is imported several times in the test suite, because each `compile_file` call will apply the configuration of `lambdapi.pkg` through `apply_config`, each of which calling `new_lib_mapping` which would fail because the mapping would have already been define by a previous compilation.

It seems better to fix your tests/ok_ko.ml and tests/utils.ml files. Instead of calling compile_file, you should better call apply_config only once by doing for instance:
```
(* ok_ko.ml *)
let _ = Files.set_lib_root (Some(".."))
let _ = Package.apply_config "OK/bool.lp"
let compile fname = Compile.compile false (Files.file_to_module fname)
...
```

Reply:
Yes, I agree indeed. But I wonder therefore whether `compile_file` is useful: we can't call it twice, so is it worth it to make function?

Reply:
> @ejgallego made some remark about this part of the code in [#434 (comment)](https://github.com/Deducteam/lambdapi/pull/434#issuecomment-712387097) . @gabrielhdt do you have a better understanding ?

No, unfortunately, I don't.